Jacoco 0.7.8 breaks tests for the new plugin mechanism

Jacoco 0.7.8 breaks the tests for the new plugin mechanism (#811) with:

```
:test

org.mockitoutil.ClassLoadersTest > cannot_load_runnable_in_given_classloader_if_some_type_cant_be_loaded FAILED
    java.lang.AssertionError: 
    Expecting a throwable with cause being an instance of:
     <java.lang.NoClassDefFoundError>
    but was an instance of:
     <java.lang.IllegalStateException: java.lang.IllegalAccessException: Class org.mockitoutil.ClassLoaders$ClassLoaderExecutor can not access a member of class org.mockitoutil.ClassLoadersTest$2 with modifiers "private static transient">
        at org.mockitoutil.ClassLoadersTest.cannot_load_runnable_in_given_classloader_if_some_type_cant_be_loaded(ClassLoadersTest.java:324)

org.mockitoutil.ClassLoadersTest > can_run_in_given_classloader FAILED
    java.lang.IllegalStateException: Given task could not be loaded properly in the given classloader 'org.mockitoutil.ClassLoadersTest$1@46dcdf84', error 'java.lang.IllegalAccessException: Class org.mockitoutil.ClassLoaders$ClassLoaderExecutor can not access a member of class org.mockitoutil.ClassLoadersTest$1 with modifiers "private static transient"

        Caused by:
        java.lang.IllegalStateException: java.lang.IllegalAccessException: Class org.mockitoutil.ClassLoaders$ClassLoaderExecutor can not access a member of class org.mockitoutil.ClassLoadersTest$1 with modifiers "private static transient"

            Caused by:
            java.lang.IllegalAccessException: Class org.mockitoutil.ClassLoaders$ClassLoaderExecutor can not access a member of class org.mockitoutil.ClassLoadersTest$1 with modifiers "private static transient"

org.mockito.internal.configuration.GlobalConfigurationTest > returns_mockito_annotation_engine_of_Plugins_if_no_MockitoConfiguration FAILED
    java.lang.IllegalStateException: Given task could not be loaded properly in the given classloader 'org.mockito.internal.configuration.GlobalConfigurationTest$1@5adb485', error 'java.lang.IllegalAccessException: Class org.mockitoutil.ClassLoaders$ClassLoaderExecutor can not access a member of class org.mockito.internal.configuration.GlobalConfigurationTest$1 with modifiers "private static transient"

        Caused by:
        java.lang.IllegalStateException: java.lang.IllegalAccessException: Class org.mockitoutil.ClassLoaders$ClassLoaderExecutor can not access a member of class org.mockito.internal.configuration.GlobalConfigurationTest$1 with modifiers "private static transient"

            Caused by:
            java.lang.IllegalAccessException: Class org.mockitoutil.ClassLoaders$ClassLoaderExecutor can not access a member of class org.mockito.internal.configuration.GlobalConfigurationTest$1 with modifiers "private static transient"

1701 tests completed, 3 failed, 42 skipped
:test FAILED
```

The corresponding Travis build: https://travis-ci.org/mockito/mockito/jobs/187043982#L431

--- a/gradle/root/coverage.gradle
+++ b/gradle/root/coverage.gradle
@@ -6,13 +6,6 @@ jacoco {
     toolVersion = "0.7.8"
 }
 
-//The tests for the new plugin mechanism are broken with Jacoco 0.7.8 - https://github.com/mockito/mockito/issues/860
-test {
-    jacoco {
-        excludes = ['org.mockitoutil.ClassLoadersTest**', 'org.mockito.internal.configuration.GlobalConfigurationTest**']
-    }
-}
-
 task mockitoCoverage(type: JacocoReport, dependsOn: "test") {
     executionData files("${buildDir}/jacoco/test.exec")
 

--- a/src/test/java/org/mockitoutil/ClassLoaders.java
+++ b/src/test/java/org/mockitoutil/ClassLoaders.java
@@ -150,11 +150,13 @@ public abstract class ClassLoaders {
                     Field declaredField = taskClassReloaded.getDeclaredField(field.getName());
                     int modifiers = declaredField.getModifiers();
                     if(Modifier.isStatic(modifiers) && Modifier.isFinal(modifiers)) {
-                        // skip otherwise IllegalAccessException (can be bypassed with Unsafe though)
+                        // Skip static final fields (e.g. jacoco fields depends on version)
+                        // otherwise IllegalAccessException (can be bypassed with Unsafe though)
                         // We may also miss coverage data.
                         continue;
                     }
-                    if (declaredField.getType() == field.getType()) { // don't copy this
+                    if (declaredField.getType() == field.getType()) { // don't copy this$0
+                        field.setAccessible(true);
                         declaredField.setAccessible(true);
                         declaredField.set(reloaded, field.get(task));
                     }

